common/options/options.go

```diff
@@ -455,6 +466,91 @@ func (o *ToolOptions) ParseArgs(args []string) ([]string, error) {
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	uriMsg := "WARNING: On some systems, a password provided directly in a connection string " +
+		"or using --uri may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	sslMsg := "WARNING: On some systems, a password provided directly using --sslPEMKeyPassword " +
+		"may be visible to system status programs such as `ps` that may be invoked by other users. " +
+		"Consider using the --config option to specify a configuration file with the password."
+
+	// Create temporary options for parsing command line args.
+	tempOpts := New("", "", EnabledOptions{Auth: true, Connection: true, URI: true})
+	_, err := tempOpts.parser.ParseArgs(args)
+	if err != nil {
+		return
+	}
+
+	// Log a message for --password, if specified.
+	if tempOpts.Auth.Password != "" {
+		log.Logvf(log.Always, passwordMsg)
+	}
+
+	// Log a message for --uri or a positional connection string, if either is specified.
+	uri := tempOpts.URI.ConnectionString
+	if uri != "" {
+		if cs, err := connstring.ParseURIConnectionString(uri); err == nil && cs.Password != "" {
+			log.Logvf(log.Always, uriMsg)
+		}
+	}
+
+	// Log a message for --sslPEMKeyPassword, if specified.
+	if tempOpts.SSL.SSLPEMKeyPassword != "" {
+		log.Logvf(log.Always, sslMsg)
+	}
+}
+
+// ParseConfigFile iterates over args to find a --config option. If not found, we return.
+// If found, we read the contents of the specified config file in YAML format. We parse
+// any values corresponding to --password, --uri and --sslPEMKeyPassword, and store them
+// in the opts.
+func (opts *ToolOptions) ParseConfigFile(args []string) error {
+	// Get config file path from the arguments, if specified.
+	_, err := opts.parser.ParseArgs(args)
+	if err != nil {
+		return err
+	}
+
+	// No --config option was specified.
+	if opts.General.ConfigPath == "" {
+		return nil
+	}
+
+	// --config option specifies a file path.
+	configBytes, err := ioutil.ReadFile(opts.General.ConfigPath)
+	if err != nil {
+		return errors.Wrapf(err, "error opening file with --config")
+	}
+
+	// Unmarshal the config file as a top-level YAML file.
+	var config struct {
+		Password          string `yaml:"password"`
+		ConnectionString  string `yaml:"uri"`
+		SSLPEMKeyPassword string `yaml:"sslPEMKeyPassword"`
+	}
+	err = yaml.UnmarshalStrict(configBytes, &config)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing config file %s", opts.General.ConfigPath)
+	}
+
+	// Assign each parsed value to its respective ToolOptions field.
+	opts.Auth.Password = config.Password
+	opts.URI.ConnectionString = config.ConnectionString
+	opts.SSL.SSLPEMKeyPassword = config.SSLPEMKeyPassword
```

Comment:
Even though v3.6 normally requires you to build with the `ssl` tag to enable the `--sslPEMKeyPassword` option, the config file would actually let you specify the password regardless.

That shouldn't break anything since the code checks for a `sslPEMKeyFile` first, so I didn't think it's necessary to check whether the tools are built with the `ssl` tag here.